File: javaswing-lab/src/main/java/org/example/common/PerformanceMetrics.java
package org.example.common;

public class PerformanceMetrics {
    
    private final double fps;
    private final double cpuUsage;
    private final long memoryUsedMB;
    private final long memoryTotalMB;
    private final long timestamp;
    
    public PerformanceMetrics(double fps, double cpuUsage, long memoryUsedMB, long memoryTotalMB) {
        this.fps = fps;
        this.cpuUsage = cpuUsage;
        this.memoryUsedMB = memoryUsedMB;
        this.memoryTotalMB = memoryTotalMB;
        this.timestamp = System.currentTimeMillis();
    }
    
    public double getFps() {
        return fps;
    }
    
    public double getCpuUsage() {
        return cpuUsage;
    }
    
    public long getMemoryUsedMB() {
        return memoryUsedMB;
    }
    
    public long getMemoryTotalMB() {
        return memoryTotalMB;
    }
    
    public long getTimestamp() {
        return timestamp;
    }
    
    @Override
    public String toString() {
        return String.format("FPS: %.1f | CPU: %.1f%% | Memory: %d/%d MB", 
                           fps, cpuUsage, memoryUsedMB, memoryTotalMB);
    }
}

File: javafx-lab/src/main/kotlin/org/example/common/PerformanceMetrics.kt
package org.example.common

data class PerformanceMetrics(
    val fps: Double,
    val cpuUsage: Double,
    val memoryUsedMB: Long,
    val memoryTotalMB: Long,
    val timestamp: Long = System.currentTimeMillis()
) {
    override fun toString(): String {
        val cpuText = if (cpuUsage < 0) "N/A" else String.format("%.1f%%", cpuUsage)
        return String.format(
            "Refresh: %.1f fps | CPU: %s | Memory: %d/%d MB",
            fps,
            cpuText,
            memoryUsedMB,
            memoryTotalMB
        )
    }
}

